CastledChess/Front
# Improve/caching

- Removed Vite PWA as it is overkill for the use we have of service workers right now
- Replaced old caching solution with a manual caching allowing us to have full control of when we cache things
- Replaced Vite PWAs service worker with a custom, simpler one -> this one only intercepts requests and serve files from cache
- Improved UX -> added a mandatory download button before using any engine.

To test this:

- test in preview mode `npm run build && npm run preview`
- go to the start-analysis page
- open your dev-tools and go to the "Application" tab
- click "Clear site data" to ensure nothing is cached before testing
- import a random pgn
- select an engine:
  - upon opening the engine select, you should see "download" icons besides the engine names, this indicates they are not cached.
  - when selecting a non-cached engine, a download button should appear besides the select. If you click it, it should download the engine and put it in the cache. The cache indicator should display in red with a few MB of data (depending on which engines you downloaded it can go to ~2MB and up to ~180MB)
  - after the download is done, it should allow you to start an analysis.
- Try downloading an engine with network throttle enabled (in the network tab of your dev-tools), it will simulate a realistic-ish connection allowing you to have an estimate of the actual user experience


As a user, I want improved caching mechanisms so that the app loads faster and responds smoothly.
Reduced loading times lead to a more seamless and enjoyable interaction with the app.
